compiler/noirc_frontend/src/tests.rs

```diff
@@ -4302,3 +4300,19 @@ fn cannot_determine_type_of_generic_argument_in_enum_constructor() {
     let features = vec![UnstableFeature::Enums];
     check_monomorphization_error_using_features(src, &features);
 }
+
+#[test]
+fn unconstrained_type_parameter_in_impl() {
+    let src = r#"
+        pub struct Foo<T> {}
+
+        impl<T, U> Foo<T> {}
+                ^ The type parameter `U` is not constrained by the impl trait, self type, or predicates
+                ~ Hint: remove the `U` type parameter
+
+        fn main() {
+            let _ = Foo {};
+        }
+        "#;
+    check_errors(src);
+}
```

Comment:
Simplified form of the original failing case:

```suggestion
}

#[test]
fn unconstrained_numeric_generic_in_impl() {
    let src = r#"
        pub struct Foo {}
        impl<let N: u32> Foo {}
                 ^ The type parameter `N` is not constrained by the impl trait, self type, or predicates
                 ~ Hint: remove the `N` type parameter
        fn main() {
            let _ = Foo {};
        }
        "#;
    check_errors(src);
}
```

Reply:
Done!